docs/guides/hooks.md

```diff
@@ -0,0 +1,85 @@
+# Hooks
+Hooks exist so that users can latch on to certain Video.js lifecycle events
+
+
+## Current Hooks
+Currently the following lifecycle events are accept and use hooks
+
+### beforesetup
+Called just before the player is created, so that the user can modify the options passed to the player or modify the video element that will be used for the player
+An example of this hook:
+
+```js
+    var beforeSetup = function(videoEl, options) {
+        videoEl.className += ' some-super-class';
+
+        // never autoplay because it makes me mad!
+        options.autoplay = false
+
+        // options that are returned here will be merged with old options
+        // automatically
+        return options;
+    };
+```
```

Comment:
I think this example (and the one below) is unclear without the `videojs.hook()` call.